R/zchunk_L103.ghg_an_USA_S_T_Y.R

```diff
@@ -34,40 +36,69 @@ module_emissions_L103.ghg_an_USA_S_T_Y_DISABLED <- function(command, ...) {
     L107.an_Prod_Mt_R_C_Sys_Fd_Y <- get_data(all_data, "L107.an_Prod_Mt_R_C_Sys_Fd_Y")
     EPA_FCCC_AG_2005 <- get_data(all_data, "emissions/EPA_FCCC_AG_2005")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are `merge` and/or 'match' calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Merge-and-Match
-    # ===================================================
+    # Map EPA ghg emissions to GCAM sector and fuel,
+    # aggregate and convert from Gg to Tg
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+    # Old system yields an emissions estimate of 0 for poultry because it has NA values.
+    L103.ghg_tg_USA_an_Sepa_F_2005 <- EPA_FCCC_AG_2005 %>%
+      left_join(EPA_ghg_tech, by = "Source_Category") %>%
+      group_by(sector, fuel) %>%
+      summarize_if(is.numeric , sum) %>%
+      filter( !is.na(sector), !is.na(fuel)) %>%
+      mutate_each( funs(replace(., is.na(.), 0))) %>%
+      mutate_if(is.numeric, funs( .*CONV_GG_TG))
+
+    } else {
+      # Corrected poltry emissions estimate, methane emissions for poultry are relatively small
+      # compared to other animals, but non-zero
+      L103.ghg_tg_USA_an_Sepa_F_2005 <- EPA_FCCC_AG_2005 %>%
+        left_join(EPA_ghg_tech, by = "Source_Category") %>%
+        group_by(sector, fuel) %>%
+        summarize_if(is.numeric, sum, na.rm = T) %>%
+        filter( !is.na(sector) , !is.na( fuel )) %>%
+        mutate_each( funs( replace(., is.na(.), 0))) %>%
+        mutate_if(is.numeric, funs( .*CONV_GG_TG) )
+
+    }
+
+    L103.ghg_tg_USA_an_Sepa_F_2005 <- EPA_FCCC_AG_2005 %>%
+      left_join(EPA_ghg_tech, by = "Source_Category") %>%
+      group_by(sector, fuel) %>%
+      summarize_if(is.numeric, sum) %>%
+      filter( !is.na(sector) , !is.na(fuel) ) %>%
+      mutate_each( funs( replace(., is.na(.), 0)) ) %>%
+      mutate_if(is.numeric, funs(.*CONV_GG_TG))
+
+    # Aggergate FAO production to EPA sectors
+    L107.an_Prod_US_Sepa_2005 <- L107.an_Prod_Mt_R_C_Sys_Fd_Y %>%
+      left_join(GCAM_sector_tech,
+                by = c('GCAM_commodity' = 'sector', 'system' = 'fuel' , 'feed' = 'technology')) %>%
+      ungroup %>%
+      mutate(year = as.numeric(year)) %>%
+      filter(year == 2005, GCAM_region_ID == 1) %>%
```

Comment:
Unclear what's happening here. Could a comment explain why we're picking out USA 2005?